### ISSUE TITLE
No wind in the game

### ISSUE THREAD

**Issue by Dany-Dan** [kirides/GD3D11#15]
Hi !

I can run successfully Gothic 2 on Linux with your directx 11 renderer.
I even found a solution to hide the mouse cursor during the game.
Everything is almost perfect, except two things.

Each time i start the game and load my save, the fov is not set correctly. I have to reload the save to fix it. Not really a big problem.

The second thing is more annoying, at least for me, because it affects the atmosphere of the game. Unlike the vanilla game, there is no wind. No grass or trees movements, nothing. I tried with others dx 11 renderers but the result is the same. 
If I press F11 and I enable numpad keys and press 1, the added grass is moving with the wind. But not the rest of the game.
Is it normal ? Is there a solution for that ?

I use Gothic 2 NOTR, gothic2_fix-2.6.0.0-rev2.exe and G2NoTR-SystemPack-1.8.exe.

**Comment by neVERberleRfellerER** [comment]
> I even found a solution to hide the mouse cursor during the game.

Please, share your knowledge.

**Comment by Dany-Dan** [comment]
> Please, share your knowledge.

I was searching for a fix and I found this alternative solution :

[Hide mouse pointer in specific window](https://www.reddit.com/r/linuxquestions/comments/elxvuh/hide_mouse_pointer_in_specific_window/)

All you need to make this script work is xdotool and mark shell script as executable.
Combined to Lutris, it works perfectly.
Note that it doesn't work for me if i put these files in .wine folder. I simply put them in a folder in my home directory.

Going back to my original question, does the wind effects are working for you with directx 11 ?

**Comment by neVERberleRfellerER** [comment]
> does the wind effects are working for you with directx 11 ?

No, I don't see any wind induced tree movement with GD3D11 in the same save that has tree moving without GD3D11.

I was going to try your mouse visibility fix just to find out, the mouse is hidden with wine-staging 5.17 - no further work necessary.

**Comment by Dany-Dan** [comment]
> No, I don't see any wind induced tree movement with GD3D11

Ok, thanks for the test !

> the mouse is hidden with wine-staging 5.17

It's weird, pointer is still visible for me with wine-staging 5.17

Unfortunately, i have been too much optimistic after starting successfully the game.
I have random black screen issue and the game freezes as soon as I use a fire spell.
I think I'm going to give up. I have already spent too much time trying to get this game to work with dx11.

**Comment by neVERberleRfellerER** [comment]
> It's weird, pointer is still visible for me with wine-staging 5.17

Interesting. maybe it's one of tkg patches (now I realize I forgot to mention I am on wine-tkg) or it's been fixed by plasma/kwin update. I'll try to pinpoint this later.

> game freezes as soon as I use a fire spell.

I can confirm this. It's related to the blast effect somehow, because when casting into the sky it does not freeze for me. Water (including ice) spells are working fine.

> I think I'm going to give up. I have already spent too much time trying to get this game to work with dx11.

Also, the ddraw performance in wine got way better this year. My main motivation for messing with DX11 renderer was extremely poor performance of ddraw renderer - for example 3 FPS in city with 40% visibility while DX11 renderer gave me 60 FPS. Or 5 second delay after every action in menu. Now I have over 50 FPS in city with ddraw renderer with max visibility and all effect enables and with enhancements from R2: AB.

**Comment by Dany-Dan** [comment]
> I can confirm this. It's related to the blast effect somehow, because when casting into the sky it does not freeze for me. Water (including ice) spells are working fine.

I found the culprit : if you press ctrl+F11 and disable Draw ParticleEffects, the game doesn't freeze anymore. But of course, all the fire effects, fire spells and other spells are not displayed anymore either...
I have read that this effect also affects Windows users by causing a freeze during teleportation.

**Comment by Kirides** [comment]
> I have read that this effect also affects Windows users by causing a freeze during teleportation.

this currently (should) only happen in Gothic 1.

also i don't know anything about WINE and it's compatibility with dll "hacks" like this

**Comment by ThielHater** [comment]
I assume this problem was solved with e3a26e48ae06487481c5355b2853d20d6d80e24f.
